src/Corax/IndexWriter.cs

```diff
@@ -1698,11 +1704,17 @@ public void Commit<TStatsScope>(TStatsScope stats)
             {
                 foreach (var (_, indexedField) in _dynamicFieldsTerms)
                 {
+                    if(indexedField.Textual.Count ==0)
+                        continue;
+                    
                     using var dynamicFieldScope = stats.For(indexedField.NameForStatistics);
 
                     using (dynamicFieldScope.For(CommitOperation.TextualValues))
-                        InsertTextualField(entriesToTermsTree, indexedField, workingBuffer, ref keys);
-
+                    {
+                        var inserter = new TextualFieldInserter(this, entriesToTermsTree, indexedField, workingBuffer);
+                        inserter.InsertTextualField();
+                        inserter.Dispose();
```

Comment:
Same question as above

Reply:
I thought that you cannot use a ref struct in try/finally. Looks like you can